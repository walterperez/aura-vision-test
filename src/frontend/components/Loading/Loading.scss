.Loading {
  width: 100vw;
  height: 100vh;
  position: absolute;
  top: 0;
  left: 0;
  display: flex;
  justify-content: center;
  align-items: center;

  .loader {
    width: 40px;
    height: 40px;
    border: 15px solid grey;
    border-top: 15px solid rgba(209, 209, 5, 1);
    border-radius: 50%;
    animation: spin 2s linear infinite;
  }
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
