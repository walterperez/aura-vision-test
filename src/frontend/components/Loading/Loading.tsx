import React from 'react';
import './Loading.scss';

export function Loading() {
  return (
    <div className="Loading" data-test="Loading">
      <div className="loader" />
    </div>
  );
}
