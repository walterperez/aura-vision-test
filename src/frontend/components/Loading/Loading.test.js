import React from 'react';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Loading } from './Loading';

configure({ adapter: new Adapter() });

describe('Loader', () => {
  it('Render Snapshot', () => {
    const output = renderer.create(<Loading />);
    expect(output).toMatchSnapshot();
  });

  it('Render', () => {
    const wrapper = shallow(<Loading />);
    const loader = wrapper.find(`[data-test="Loading"]`);
    expect(loader.length).toBe(1);
  });
});
