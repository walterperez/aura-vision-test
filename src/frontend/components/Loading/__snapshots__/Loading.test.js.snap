// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Render Snapshot 1`] = `
<div
  className="Loading"
  data-test="Loading"
>
  <div
    className="loader"
  />
</div>
`;
